File: src/main/java/io/github/starwishsama/namelessbot/objects/GroupShop.java
package io.github.starwishsama.namelessbot.objects;

import io.github.starwishsama.namelessbot.BotConstants;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.atomic.AtomicReference;

@Data
public class GroupShop {
    private long groupId;
    private List<BotUser> admins;
    private ArrayList<ShopItem> items;

    public GroupShop() {

    }

    public GroupShop(Long groupId) {
        this.groupId = groupId;
    }

    public ShopItem getItemByName(String itemName) {
        AtomicReference<ShopItem> result = new AtomicReference<>(new ShopItem());
        items.forEach(item -> {
            if (item.getItemName().equals(itemName)) {
                result.set(item);
            }
        });
        return result.get();
    }

    public void addAdmin(BotUser user) {
        admins.add(user);
    }

    public void removeAdmin(BotUser user) {
        admins.remove(user);
    }

    public void addNewItem(ShopItem item) {
        if (!items.contains(item)) {
            items.add(item);
        }
    }

    public static GroupShop getShopById(Long groupId) {
        AtomicReference<GroupShop> result = new AtomicReference<>(new GroupShop());
        BotConstants.shop.forEach(shop -> {
            if (shop.getGroupId() == groupId) {
                result.set(shop);
            }
        });
        return result.get();
    }
}

File: src/main/kotlin/io/github/starwishsama/nbot/objects/GroupShop.kt
package io.github.starwishsama.nbot.objects

import io.github.starwishsama.nbot.BotConstants
import java.util.*
import java.util.concurrent.atomic.AtomicReference
import java.util.function.Consumer


class GroupShop {
    var groupId: Long = 0
    val admins: MutableList<BotUser>? = null
    var items: ArrayList<ShopItem?>? = ArrayList()

    constructor() {}

    constructor(groupId: Long) {
        this.groupId = groupId
    }

    fun getItemByName(itemName: String): ShopItem? {
        val result = AtomicReference<ShopItem?>()
        items!!.forEach(Consumer { item: ShopItem? ->
            if (item!!.itemName == itemName) {
                result.set(item)
            }
        })
        return result.get()
    }

    fun addAdmin(user: BotUser) {
        admins!!.add(user)
    }

    fun removeAdmin(user: BotUser) {
        admins!!.remove(user)
    }

    fun addNewItem(item: ShopItem) {
        if (items == null) {
            items = ArrayList()
        }
        if (!items!!.contains(item)) {
            items!!.add(item)
        }
    }

    companion object {
        fun getShopById(groupId: Long): GroupShop {
            val result = AtomicReference(GroupShop())
            BotConstants.shop.forEach { shop ->
                if (shop.groupId == groupId) {
                    result.set(shop)
                }
            }
            return result.get()
        }
    }
}
